### PR TITLE
engine: Set full state on settings struct

### PR DESCRIPTION
**Pull request by shazbert**
# PR Description

Just noticed a potential slight improvement in the engine. Let me know what you guys think or if there are any problems with this.

Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: engine/engine.go [hunk 1]
```diff
@@ -137,17 +137,10 @@ func loadConfigWithSettings(settings *Settings, flagSet map[string]bool) (*confi
 
 // validateSettings validates and sets all bot settings
 func validateSettings(b *Engine, s *Settings, flagSet map[string]bool) {
-	b.Settings.Verbose = s.Verbose
-	b.Settings.EnableDryRun = s.EnableDryRun
-	b.Settings.EnableAllExchanges = s.EnableAllExchanges
-	b.Settings.EnableAllPairs = s.EnableAllPairs
-	b.Settings.EnableCoinmarketcapAnalysis = s.EnableCoinmarketcapAnalysis
-	b.Settings.EnableDatabaseManager = s.EnableDatabaseManager
+	// Set full state from command line
```
Comment (gloriousCode) [inline engine/engine.go:140]: I don't think this comment is needed

**Review by gloriousCode — APPROVED** [review]
tACK!

I've compared the two structs before and after to ensure no unintended settings are being messed with, and it all checks out.

_HOWEVER_, your PR has highlighted that we were never setting the following fields:
```
	EnableCurrencyConverter bool
	EnableCurrencyLayer     bool
	EnableFixer             bool
	EnableOpenExchangeRates bool
	EnableExchangeRateHost  bool
```

To demonstrate, you can run the following command on your branch and master branch.
`go run . --openexchangerates=true`

On master branch GCT output:
` Enable OpenExchangeRates: false`
Now on your branch GCT output:
`Enable OpenExchangeRates: true`

So thank you for fixing additional bugs, even if you didn't mean to 😄

**Review by thrasher- — APPROVED** [review]
Nice simplification, tACK once Scott's comment nit is addressed 👍

File: engine/engine.go (hunk 1 above)
Comment (shazbert) [reply engine/engine.go]: done!

**Review by MadCozBadd — APPROVED** [review]
tACK!

**Review by gloriousCode — APPROVED** [review]
Thanks for making that change 
# 🎉

File: engine/engine.go [hunk 2]
```diff
@@ -189,9 +181,6 @@ func validateSettings(b *Engine, s *Settings, flagSet map[string]bool) {
 		withdraw.CacheSize = s.WithdrawCacheSize
 	}
 
-	b.Settings.EnableCommsRelayer = s.EnableCommsRelayer
-	b.Settings.EnableEventManager = s.EnableEventManager
-
 	if b.Settings.EnableEventManager {
 		if b.Settings.EventManagerDelay != time.Duration(0) && s.EventManagerDelay > 0 {
```
Comment (gloriousCode) [inline engine/engine.go:185]: This makes a little less sense now actually. b.Settings is s now.

**Review by gloriousCode — CHANGES_REQUESTED** [review]
See my latest comment about the if statements. Its mostly harmless, but its a little weird logic to follow along with to set a value to itself

File: engine/engine.go [hunk 3]
```diff
@@ -137,99 +137,52 @@ func loadConfigWithSettings(settings *Settings, flagSet map[string]bool) (*confi
...
 	if b.Settings.EnablePortfolioManager {
-		if b.Settings.PortfolioManagerDelay == time.Duration(0) && s.PortfolioManagerDelay > 0 {
-			b.Settings.PortfolioManagerDelay = s.PortfolioManagerDelay
-		} else {
+		if b.Settings.PortfolioManagerDelay <= 0 {
 			b.Settings.PortfolioManagerDelay = PortfolioSleepDelay
 		}
 	}
 
 	if flagSet["grpc"] {
```
Comment (gloriousCode) [inline engine/engine.go:151]: Invert this `if` please.

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Thanks for making that change! One more!

File: engine/engine.go (hunk 3 above)
Comment (shazbert) [reply engine/engine.go:151]: 🤭 okay

**Review by gloriousCode — APPROVED** [review]
tACK! Thank you for making that change!

**Review by MadCozBadd — APPROVED** [review]
Awesome stuff! New changes tested and approved!

**Review by thrasher- — APPROVED** [review]

**Pull request merged**